app/models/group_assignment_invitation.rb

```diff
@@ -29,9 +29,11 @@ class GroupAssignmentInvitation < ApplicationRecord
   def redeem_for(invitee, selected_group = nil, new_group_title = nil)
     return Result.failed("Invitations for this assignment have been disabled.") unless enabled?
 
-    repo_access    = RepoAccess.find_or_create_by!(user: invitee, organization: organization)
-    invitees_group = group(repo_access, selected_group, new_group_title)
+    repo_access = RepoAccess.find_or_create_by!(user: invitee, organization: organization)
+    group_creator_result = group(repo_access, selected_group, new_group_title)
+    return Result.failed(group_creator_result.error) if group_creator_result.failed?
```

Comment:
What happens in this case, is the error caught further up the stack? Just checking because we didn't return a `Result.failed` in any case before

Reply:
Previously if it failed to create a group it would throw a `GitHub::Error` here (and it wouldn't be handled 😬).
With this change we handle handle group creation errors in the `Group::Creator`:
https://github.com/education/classroom/blob/383df517636d446c8a25c6f6e8cebc61ccbbffd4/app/models/group/creator.rb#L60-L61